pkg/controller/control.go

```diff
@@ -694,6 +696,15 @@ func determineCorrectVolumeSize(volumeSize, volumeCurrentSize int64, backendSize
 	return volumeCurrentSize
 }
 
+func isBackendServiceUnavailable(errorCodes map[string]codes.Code) bool {
+	for _, code := range errorCodes {
+		if code == codes.Unavailable {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
For this implementation, we will not regard all replicas failed if there is at least one is unable to reach. It seems not granular enough due to the current design limitation, but ideally we should be able to determine each replica individually. 

My question is shouldn't we check all error codes of every replica instead of only one?

Reply:
> My question is shouldn't we check all error codes of every replica instead of only one?

I think it should be fine because individual replica should be able to sync their status from the instance manager. 

![image](https://user-images.githubusercontent.com/899492/225821545-d20f38cc-0654-4a74-9577-fe96ff97625e.png)

Reply:
@derekbit Can you confirm this?

Reply:
If one or more than one of the replicas failed caused by service unavailable, the all replicas won't be marked as error.
The volume controller will retry the attachment. Once it succeeds, the engine monitor will set the real failed replicas to ERR state, and later the volume controller marked them as failed.

Reply:
Is it possible to become endless retry?

Reply:
Do you mean if the replicas' services are always unavailable? If yes, it will be at attach/detach loop.

Reply:
Yeah but it seems fair because it's just the result due to unexpected environment situation.

Reply:
Yeah, I think the behavior looks reasonable.